tests/bases/test_metric.py

```diff
@@ -423,3 +423,18 @@ class UnsetProperty(metric_class):
         match="Torchmetrics v0.9 introduced a new argument class property called.*",
     ):
         UnsetProperty()
+
+
+@pytest.mark.parametrize("metric_class", [DummyListMetric, DummyMetric, DummyMetricMultiOutput, DummyMetricSum])
+def test_no_warning_on_custom_forward(metric_class):
+    """If metric is using custom forward, full_state_update is irrelevant."""
+
+    class UnsetProperty(metric_class):
+        full_state_update = None
+
+        def forward(self, *args, **kwargs):
+            self.update(*args, **kwargs)
+
+    with pytest.warns(None) as record:
+        UnsetProperty()
+    assert len(record) == 0
```

Comment:
We have a nicer utility for testing the lack of warnings in PL: https://github.com/PyTorchLightning/pytorch-lightning/blob/7938293cd9bb27e5ad97aabc7261fb29027dc6b3/tests/helpers/utils.py#L118 in case you want to copy and use it.

Reply:
thanks :]